{{cookiecutter.project_name}}/meson-mesonpy.build

```diff
@@ -0,0 +1,40 @@
+project(
+  '{{ cookiecutter.project_name.replace("-", "_") }}',
+  'cpp',
+  version: '0.1.0',
+  license: '{{ cookiecutter.license }}',
+  meson_version: '>= 0.64.0',
+  default_options: [
+    'buildtype=debugoptimized',
+    'cpp_std=c++11',
+  ],
+)
+name = '{{ cookiecutter.project_name.replace("-", "_") }}'
+
+py_mod = import('python')
+py = py_mod.find_installation(pure: false)
+
+pybind11_config = find_program('pybind11-config')
+pybind11_config_ret = run_command(pybind11_config, ['--includes'], check: true)
+pybind11 = declare_dependency(
+    include_directories: [pybind11_config_ret.stdout().split('-I')[-1].strip()],
+)
+py.install_sources(
```

Comment:
There are two schools of thought when it comes to installing files. One is the "detect a source tree" approach (e.g. setuptools `find_packages()`), and one is the "explicitly list all files" approach.

This is the latter approach. The former approach can be accomplished in meson with:

```meson
install_subdir('src', install_dir: py.get_install_dir())
```

Without wearing my meson hat... I personally recommend listing all sources explicitly, guesswork is too often an issue especially when you e.g. forget to check new files into git, and then both your local tree and the version you pushed to a public software forge build successfully, but do different things.

Reply:
Yup, and meson is committed to supporting that use case, even if some of its developers wouldn't personally use it. :)